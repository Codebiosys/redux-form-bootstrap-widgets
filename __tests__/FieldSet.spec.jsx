/* eslint-disable import/first */

import React from 'react';
import { shallow } from 'enzyme';
import { FieldSet } from 'index';

const fieldProps = {
  label: 'Field Label',
};

describe('The Field Set', () => {
  let inputWrapper;

  beforeEach(() => {
    inputWrapper = shallow(<FieldSet {...fieldProps}><input name="foo" /></FieldSet>);
  });

  it('renders', () => {
    expect(inputWrapper).toMatchSnapshot();
  });

  it('has a label when there is a label', () => {
    expect(inputWrapper.find('legend').text()).toEqual(fieldProps.label);
  });
});
