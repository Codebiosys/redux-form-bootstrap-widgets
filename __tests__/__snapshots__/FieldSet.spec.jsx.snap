// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Field Set renders 1`] = `
<fieldset
  className="form-group"
  style={
    Object {
      "border": "1px solid #e5e5e5",
      "margin": "0 2px",
      "padding": ".35em .625em .75em",
    }
  }
>
  <legend
    style={
      Object {
        "borderBottom": "0px",
        "fontSize": "1.5em",
        "paddingLeft": "10px",
        "paddingRight": "10px",
        "width": "auto",
      }
    }
  >
    Field Label
  </legend>
  <input
    name="foo"
  />
</fieldset>
`;
